api/v1alpha4/awscluster_types.go

```diff
@@ -0,0 +1,184 @@
+	// +optional
+	ControlPlaneLoadBalancer *AWSLoadBalancerSpec `json:"controlPlaneLoadBalancer,omitempty"`
+
+	// ImageLookupFormat is the AMI naming format to look up machine images when
+	// a machine does not specify an AMI. When set, this will be used for all
+	// cluster machines unless a machine specifies a different ImageLookupOrg.
+	// Supports substitutions for {{.BaseOS}} and {{.K8sVersion}} with the base
+	// OS and kubernetes version, respectively. The BaseOS will be the value in
+	// ImageLookupBaseOS or ubuntu (the default), and the kubernetes version as
+	// defined by the packages produced by kubernetes/release without v as a
+	// prefix: 1.13.0, 1.12.5-mybuild.1, or 1.17.3. For example, the default
+	// image format of capa-ami-{{.BaseOS}}-?{{.K8sVersion}}-* will end up
+	// searching for AMIs that match the pattern capa-ami-ubuntu-?1.18.0-* for a
+	// Machine that is targeting kubernetes v1.18.0 and the ubuntu base OS. See
+	// also: https://golang.org/pkg/text/template/
+	// +optional
+	ImageLookupFormat string `json:"imageLookupFormat,omitempty"`
+
+	// ImageLookupOrg is the AWS Organization ID to look up machine images when a
+	// machine does not specify an AMI. When set, this will be used for all
+	// cluster machines unless a machine specifies a different ImageLookupOrg.
+	// +optional
+	ImageLookupOrg string `json:"imageLookupOrg,omitempty"`
+
+	// ImageLookupBaseOS is the name of the base operating system used to look
+	// up machine images when a machine does not specify an AMI. When set, this
+	// will be used for all cluster machines unless a machine specifies a
+	// different ImageLookupBaseOS.
+	ImageLookupBaseOS string `json:"imageLookupBaseOS,omitempty"`
+
+	// Bastion contains options to configure the bastion host.
+	// +optional
+	Bastion Bastion `json:"bastion"`
+}
+
+type Bastion struct {
+	// Enabled allows this provider to create a bastion host instance
+	// with a public ip to access the VPC private network.
+	// +optional
+	Enabled bool `json:"enabled"`
+
+	// DisableIngressRules will ensure there are no Ingress rules in the bastion host's security group.
+	// Requires AllowedCIDRBlocks to be empty.
+	// +optional
+	DisableIngressRules bool `json:"disableIngressRules,omitempty"`
+
+	// AllowedCIDRBlocks is a list of CIDR blocks allowed to access the bastion host.
+	// They are set as ingress rules for the Bastion host's Security Group (defaults to 0.0.0.0/0).
+	// +optional
+	AllowedCIDRBlocks []string `json:"allowedCIDRBlocks,omitempty"`
+
+	// InstanceType will use the specified instance type for the bastion. If not specified,
+	// Cluster API Provider AWS will use t3.micro for all regions except us-east-1, where t2.micro
+	// will be the default.
+	InstanceType string `json:"instanceType,omitempty"`
+
+	// AMI will use the specified AMI to boot the bastion. If not specified,
+	// the AMI will default to one picked out in public space.
+	// +optional
+	AMI string `json:"ami,omitempty"`
```

Comment:
Same comment as above.
Did not do any changes on the v1alpha4 types.
Let's track the changes we want to have in v1alpha4 as separate issues so that it will be easier to review/revert.